docs/compilers/Design/Bound Node Design.md

```diff
@@ -0,0 +1,15 @@
+﻿Bound Node Design
+=================
+
+This document discusses design principles for the bound nodes.
+
+### The shape of the bound tree should correspond to the shape of the program's static semantics
+
+Generally speaking, that means that there is an isomorphism between the syntax and bound nodes, except when there is a mismatch between the shape of the two.  When possible, we prefer the correspondence be as direct as possible.  Here are two examples that illustrate this:
```

Comment:
nit: "except when there is a mismatch between the shape of the two" seems redundant

Reply:
Clarifying.  I meant when there is a mismatch between the syntax and semantics.

---
In reply to: [359572054](https://github.com/dotnet/roslyn/pull/40457#discussion_r359572054) [](ancestors = 359572054)